iroha/src/sumeragi.rs

```diff
@@ -108,28 +110,32 @@ impl Sumeragi {
                 Ok(Some(block))
             } else {
                 self.voting_block = Some(block.clone());
-                let mut send_futures = Vec::new();
+                let torii_sender = self.messages_to_torii_sender.write().await;
                 for peer in self.validating_peers() {
-                    send_futures.push(Message::Created(block.clone()).send_to(peer));
+                    torii_sender
+                        .send(Message::Created(block.clone()).as_torii_message(peer))
+                        .await;
                 }
-                send_futures.push(Message::Created(block.clone()).send_to(self.proxy_tail()));
-                futures::future::join_all(send_futures).await;
+                torii_sender
+                    .send(Message::Created(block.clone()).as_torii_message(self.proxy_tail()))
+                    .await;
                 Ok(None)
             }
         } else {
             //TODO: send pending transactions to all peers and as leader check what tx have already been committed
             //Sends transactions to leader
-            let mut send_futures = Vec::new();
+            let torii_sender = self.messages_to_torii_sender.write().await;
             for transaction in &transactions {
-                send_futures.push(Network::send_request_to(
-                    &self.leader().address,
-                    Request::new(
-                        uri::INSTRUCTIONS_URI.to_string(),
-                        RequestedTransaction::from(transaction).into(),
-                    ),
-                ));
+                torii_sender
```

Comment:
Torii handles incoming requests, but doesn't need to be an abstraction about outgoing network requests.